bundles/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/internal/handler/SomfyTahomaBaseThingHandler.java

```diff
@@ -387,12 +393,7 @@ public void updateThingStatus(List<SomfyTahomaState> states) {
     }
 
     private @Nullable SomfyTahomaState getStatusState(List<SomfyTahomaState> states) {
-        for (SomfyTahomaState state : states) {
-            if (STATUS_STATE.equals(state.getName()) && state.getType() == TYPE_STRING) {
-                return state;
-            }
-        }
-        return null;
+        return SomfyTahomaDevice.getState(states, STATUS_STATE, TYPE_STRING);
```

Comment:
Please see below. I do not want to move the binding logic to model classes.

Reply:
Ok, I moved the new method `getState` into `SomfyTahomaBaseThingHandler`.